tests/test_badfiles.py

```
@@ -595,3 +595,54 @@ def test_extract_rejects_symlink_to_archive_itself(tmp_path):
+    )
+
+    with pytest.raises(Bad7zFile):
+        SevenZipFile(io.BytesIO(data))
+
+
+@pytest.mark.misc
+def test_append_to_corrupt_archive_does_not_overwrite():
+    """Opening a corrupt 7z file in append mode must raise rather than fall
+    through to write mode, which would overwrite the file."""
+    import base64
+    import io
```

Flake8 (7.3.0)

[error] 'io' imported but unused

(F401)